tests/solana_tests/simple.rs

```diff
@@ -581,3 +581,110 @@ contract test3 {
         );
     }
 }
+
+#[test]
+fn overloading() {
+    let mut vm = build_solidity(
+        r#"
+        contract c {
+            function foo() public returns (uint32) { return 1; }
+            function foo_bar() public returns (uint32) { return 0; }
+            function foo(address) public returns (uint32) { return 2; }
+            function foo(bytes32) public returns (uint32) { return 3; }
+            function foo(bytes) public returns (uint32) { return 4; }
+            function foo(string) public returns (uint32) { return 5; }
+        }"#,
+    );
+
+    let data_account = vm.initialize_data_account();
+    vm.function("new")
+        .accounts(vec![("dataAccount", data_account)])
+        .call();
+
+    let returns = vm
+        .function("foo_")
```

Comment:
What happens when I call only `foo` (without the underscore)?

Reply:
That does not work, the mangling appends `_` followed by the argument types for overloaded functions. In  this case, there are no arguments so the function is called `foo_`.  See https://github.com/hyperledger/solang/blob/main/src/sema/ast.rs#L422

Reply:
So `"("` => `"_"` and `")"` => '"", that's how we end up there.